### PR TITLE
PCHR-2753: Tasks filter by date doesn't work on 1.6 and 1.7

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
After adding fix in PR #245 and in course of adopting angular js style guide, there left few regression issues causing the filters in tasks broken:
issues:
1. https://github.com/compucorp/civihr-tasks-assignments/pull/245/files#diff-9ddc2c346e6840e0dfe2f4c8d2217fb3R96
2. https://github.com/compucorp/civihr-tasks-assignments/pull/245/files#diff-d034a1547714f392d984ef97994dd244R34
3. https://github.com/compucorp/civihr-tasks-assignments/pull/245/files#diff-d034a1547714f392d984ef97994dd244R48

This PR fixes those introduced/unnoticed issues both in `/tasks` and `/documents`

## Before
1. The task filters were broken.
datetasks

2. Filtering the tasks when clicking the tabs in `/tasks` not working with correct counter value.
tasksandfilters

## After
1. The tasks filters work properly.
2572-after-min

2. Filtering the tasks when clicking the tab in `/tasks` not works with correct counter value.
2753-ffix-iloveimg-compressed

3. Filtering the Documents when clicking the tab in `/documents` works with correct counter value.
2753-fix-plus-min

## Technical Details
1. Use defined function `filterByDateField()` for filter tasks by date range in task list in file `/views/dashboard/tasks.html`
```html
<li ng-repeat="task in listFiltered = (listUnlimited = (listOngoing = (list.filterByDateField(list.filterParams.due) ...) | limitTo: listLimit)"
```
2. Remove unnecessary code in sidebar filters, form date picker fields. Which were breaking filters in Task list. These changes were already removed in [commit](https://github.com/compucorp/civihr-tasks-assignments/commit/2dd8969072bc78d85d0448815aab595679a885cf#diff-d034a1547714f392d984ef97994dd244) in file `/views/directives/sidebar-filters.html`.

```html
..
ng-click="list.dpOpened.filterDates.from = !list.dpOpened.filterDates.from"
..
ng-click="list.dpOpened.filterDates.until = !list.dpOpened.filterDates.until"
..
```
3. Use the ``filterByDateField()`` function to filter tasks by date range in tabbed task filters in `/tasks` in T&A Dashboard (file: /views/dashboard/tasks.html)
```html
<div class="col-xs-12 col-sm-6 text-center">
    ...
    My Tasks
    <span>
        ({{(list.filterByDateField(list.filterParams.due) | filterByStatus:cache.taskStatusResolve:false | filterByOwnership:'assigned').length}})
    </span>
    ...
    Delegated Tasks
    <span>
       ({{(list.filterByDateField(list.filterParams.due) | filterByStatus:cache.taskStatusResolve:false | filterByOwnership:'delegated').length}})
    </span>
    ...
    All
    <span>
        ({{(list.filterByDateField(list.filterParams.due) | filterByStatus:cache.taskStatusResolve:false).length}})
   ...
</div>
```
4. Use function `filterByDateField()` in "views/dashboard/documents.html"
```html
<a ... 
    All
    <span>
        ({{
           ((list.filterByDateField(list.filterParams.due)) | filterByStatus:list.filterParams.documentStatus).length
         }})
     </span>
</a>
```
## For QA:
1. Verify that the tasks can be filtered in `/tasks` in T&A:
- Over Due
- Due Today
- Due This week

2. Verify that the filtering tasks works with correct counter by clicking following  in `/tasks`.
- My Tasks
- Delegated Tasks
- All

3. Verify that the filtering documents works with correct counter by clicking following in `/dcuments`: 
- My Docs
- Delegated Docs
- All

4. Verify the tasks can be filtered using custom date ranges 
5. Verify that filtering documents in `/documents` path in T&A still works 

## Comments
- Verified that the task filters work as expected.
- Also verified the document filters work as expected.

## Tests
✅ Jasmine Tests - passed
⏹ JS distributive files - not needed
⏹ Backstop Tests - n/a
⏹ PHP Unit Tests - not needed to be run
⏹ CSS distributive files - not needed

### CODE REVIEW

**Comment by igorpavlov** [comment]
I am not sure which exact tests were passed. Please have a look at this PR overview and let me know if you like the decoration of passed tests: https://github.com/civicrm/civihr/pull/2205

I think we might try to organise like that:

- ✅ JS distributive files - included
- ⏹ CSS distributive files - not needed
- ✅ Jasmine Tests - passed
- ❌ Backstop Tests - failed
- ⚠️ PHP Unit Tests - not performed

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html [hunk 1]
```diff
@@ -92,8 +91,15 @@ <h3 class="panel-title">{{list.label[list.filterParams.due]}}</h3>
           </div>
         </div>
         <ul class="{{prefix}}list-task">
-          <li ng-repeat="task in listFiltered = (listUnlimited = (listOngoing = (list.list | filterByStatus:cache.taskStatusResolve:false) | filterByDateField:list.filterParams.due:list.filterParams.dateRange | filterByOwnership:list.filterParams.ownership | filterByAssignmentType:list.filterParams.assignmentType | filterByContactId:list.filterParams.contactId | orderBy: 'activity_date_time') | limitTo: listLimit)"
-            ng-controller="TaskCtrl" ng-class="{ '{{prefix}}task-completed': task.completed, '{{prefix}}task-due': task.due }" ng-include src="'task.html'" ct-spinner ct-spinner-id="task{{task.id}}">
+          <li ng-repeat="task in listFiltered = (listUnlimited = (listOngoing = (list.filterByDateField(list.filterParams.due)
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: I would ask to organise this code a bit differently, if you don't mind:

```html
<li ng-repeat="task in listFiltered = 
  (listUnlimited = 
    (listOngoing = 
      (list.list | filterByStatus:cache.taskStatusResolve:false) 
      | filterByDateField:list.filterParams.due:list.filterParams.dateRange 
      | filterByOwnership:list.filterParams.ownership 
      | filterByAssignmentType:list.filterParams.assignmentType 
      | filterByContactId:list.filterParams.contactId 
      | orderBy: 'activity_date_time'
    ) | limitTo: listLimit)"
  ng-controller="TaskCtrl"
  ng-class="{ '{{prefix}}task-completed': task.completed, '{{prefix}}task-due': task.due }"
  ng-include src="'task.html'"
  ct-spinner
  ct-spinner-id="task{{task.id}}">
```

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: Looks better! will do.

Comment (igorpavlov) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: Could you please indent what's inside ng-repeat by 2 spaces? It looks a bit odd that <li> and (listUnlimited have the same indentation level. Please have a look at my snippet ^^

**Comment by igorpavlov** [comment]
@Buddha, as per tests, I meant including **all** possible tests that can be performed and specifying if they were passed, failed, not needed or simply ignored. Otherwise, the only difference if a fancy emoji icon vs grey native checkbox.

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html (hunk 1 above)
Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: Fixed!

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html [hunk 2]
```diff
@@ -70,8 +70,7 @@ <h3 class="panel-title">{{list.label[list.filterParams.due]}}</h3>
                 All
               <span>
                 ({{
-                  (list.list |
-                  filterByDateField:list.filterParams.due:list.filterParams.dateRange |
+                  ((list.filterByDateField(list.filterParams.due)) |
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: You do not need `(list.filterByDateField(list.filterParams.due))`, you can do `list.filterByDateField(list.filterParams.due)`

**Pull request merged**